qemu/tests/virtio_console.py

```diff
@@ -65,7 +65,8 @@ def add_chardev(vm, params):
                                     'unix_socket')
         if backend in ['udp', 'tcp_socket']:
             chardev_param['chardev_host'] = host
-            chardev_param['chardev_port'] = str(free_ports[index])
+            _port = chardev_param.get('chardev_port', str(free_ports[index]))
+            chardev_param['chardev_port'] = _port
```

Comment:
With the new solution, we do not need to update this function now.

Reply:
but we still need this function, because if it doesnt get 'chardev_port' it will auto assign another free port number

Reply:
With the new solution,  it only call `add_chardev` once for the first vm.
For the second vm, just update the device param with one new option "to" is ok.